### PR TITLE
docs(providers): use baseprovider in the docs

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: documentation-site/pages/theming/custom-themes.mdx [hunk 1]
```diff
@@ -35,14 +35,14 @@ All themeable values are available within components that provide the `overrides
 
 To learn more on how overrides work, check out [Better Reusable React Components with the Overrides Pattern](https://medium.com/@dschnr/better-reusable-react-components-with-the-overrides-pattern-9eca2339f646) article.
 
-## The `ThemeProvider`
+## The `BaseProvider`
 
-With the `ThemeProvider`, you can apply a theme to your application globally or locally. By default, Base Web ships with a light and a dark theme.
+With the `BaseProvider`, you can apply a theme to your application globally or locally. By default, Base Web ships with a light and a dark theme.
```
Comment (nadiia) [inline documentation-site/pages/theming/custom-themes.mdx]: How about changing it in a way that says `use BaseProvider to apply a theme globally and ThemeProvider to apply it locally`?

File: documentation-site/pages/theming/custom-themes.mdx [hunk 2]
```diff
@@ -57,19 +57,19 @@ export default function App() {
 ```
 
 If you'd like to use a different theme in one part of your application, \
-you can locally wrap that part with the ThemeProvider to override the theme locally:
```
Comment (nadiia) [inline documentation-site/pages/theming/custom-themes.mdx]: In this part of the docs the `ThemeProvider` usage would be the correct recommendation.

File: documentation-site/pages/theming/custom-themes.mdx (hunk 1 above)
Comment (gergelyke) [reply documentation-site/pages/theming/custom-themes.mdx]: is there a difference between the two? if not, I think we should tell folks to use the same thing everywhere to keep things simple

File: documentation-site/pages/theming/custom-themes.mdx (hunk 2 above)
Comment (gergelyke) [reply documentation-site/pages/theming/custom-themes.mdx]: would it work with the BaseProvider too? if so, I think we should let folks know only one way to do it, so it won't be confusing

File: documentation-site/pages/theming/custom-themes.mdx (hunk 1 above)
Comment (nadiia) [reply documentation-site/pages/theming/custom-themes.mdx]: The difference is that BaseProvider adds more Providers than a single Theme one. We don't want another LayersProvider to be added in a separate part of the app, unless it's intended.

Comment (gergelyke) [reply documentation-site/pages/theming/custom-themes.mdx]: oh, that makes sense - let me update this PR

**Comment by gergelyke** [comment]
thanks for the comments @nadiia  - just updated the PR, please let me know what you think

**Pull request merged**